src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs

```diff
@@ -858,10 +875,15 @@ private static IEnumerable<InstalledPackageReference> GetPackageReferencesFromTa
             newProject.Build(new[] { CollectPackageReferences }, new List<Microsoft.Build.Framework.ILogger> { }, out var targetOutputs);
 
             return targetOutputs.First(e => e.Key.Equals(CollectPackageReferences, StringComparison.OrdinalIgnoreCase)).Value.Items.Select(p =>
-                new InstalledPackageReference(p.ItemSpec)
+            {
+                var isVersionOverride = p.GetMetadata("VersionOverride") != string.Empty;
```

Comment:
>VersionOverride [](http://example.com/codeflow?start=55&length=15)

I'm a bit concerned that these properties are not defined centrally somewhere.

Reply:
I created the issue https://github.com/NuGet/Home/issues/12865, assigned to myself for next quality week.